src/fstab-generator/fstab-generator.c

```diff
@@ -358,6 +358,15 @@ static int add_mount(
                 "Documentation=man:fstab(5) man:systemd-fstab-generator(8)\n",
                 source);
 
+        if (strneq(fstype, "nfs", 3) && !automount &&
+            fstab_test_yes_no_option(opts, "bg\0" "fg\0")) {
+                opt_bg = strjoin("x-systemd.mount-timeout=infinity,retry=10000,", opts, ",fg");
+                if (!opt_bg)
+                        return log_oom();
```

Comment:
given that the string is bounded, it might be nicer to use strjoina() here, i.e. alloca() memory, so that we can't fail here, and don't need to bother with OOM, or freeing...